File: javascript/exercise08/pizza-party.js
import { input } from "pynput";

const people = Number(await input("How many people? "));
if (!people) {
	console.log("Please enter a valid number!!");
	process.exit();
}

const pizzas = Number(await input("How many pizzas do you have? "));
if (!pizzas) {
	console.log("Please enter a valid number!!");
	process.exit();
}

const leftovers = people % pizzas;
console.log(
	`${people} people with ${pizzas} pizzas\nEach person gets ${people / pizzas / 2} pieces of pizza.`,
);
console.log(`There are ${leftovers} leftovers pieces.`);

File: go/pizza_party.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)

	fmt.Print("How many people? ")
	scanner.Scan()
	people, err := strconv.Atoi(scanner.Text())
	if err != nil {
		fmt.Println("Please enter a valid number!!")
		return
	}

	fmt.Print("How many pizzas do you have? ")
	scanner.Scan()
	pizzas, err := strconv.Atoi(scanner.Text())
	if err != nil {
		fmt.Println("Please enter a valid number!!")
		return
	}

	leftovers := people % pizzas
	fmt.Printf("%v people with %v pizzas\nEach person gets %v pieces of pizza.\n", people, pizzas, (people/pizzas)/2)
	fmt.Printf("There are %v leftovers pieces.\n", leftovers)

}
